pkg/apis/auditregistration/validation/validation.go

```diff
@@ -48,6 +51,10 @@ func ValidateWebhook(w auditregistration.Webhook, fldPath *field.Path) field.Err
 		allErrs = append(allErrs, ValidateWebhookThrottleConfig(w.Throttle, fldPath.Child("throttle"))...)
 	}
 
+	for _, version := range w.EventVersions {
+		allErrs = append(allErrs, ValidateEventVersion(version, fldPath.Child("eventVersion"))...)
```

Comment:
this requires all versions to be recognized by the apiserver, which is not quite what we want

1. this is called on update, which means a failure here could prevent update of an existing persisted object
2. requiring recognition of all items in the list means a sink couldn't register for `v1beta1,v1` against a server which only supports `v1beta1`. This scenario is main reason to accept multiple versions.

what we want is:
1. on create, require at least one recognized version
2. on update, if the existing object has at least one recognized version, require the new object to as well

then, when sending events to the audit sink, iterate through the accepted versions, sending the first one we know how to handle

Reply:
updated
> 2. on update, if the existing object has at least one recognized version, require the new object to as well

I simply handled update the same as create, requiring a recognized version

Reply:
> I simply handled update the same as create, requiring a recognized version

That's unsafe in HA skew and rollback scenarios where the stored data was persisted via a server that supported audit event versions the current server doesn't, and only claimed support for those versions

Reply:
> on update, if the existing object has at least one recognized version, require the new object to as well
could you clarify what should happen if the old object does not have a supported version by a new server? Wouldn't we still want to verify the new object does, and replace the old one?

Reply:
> could you clarify what should happen if the old object does not have a supported version by a new server? Wouldn't we still want to verify the new object does, and replace the old one?

if the existing persisted object does not have a supported version (for whatever reason), we should not require the new object to either. this allows unrelated updates (labeling, removing finalizers, updating CA bundle, etc)

Reply:
the validation in https://github.com/kubernetes/kubernetes/pull/74998/commits/eb025089ff855de403f646aed1973bcd560fdc26#diff-54fdd67617b0f2ef1b5a09148f77303eR116 is a good example of what we're going for